Homepage Task List [MAILPOET-4826]

costasovo opened the pull request:
## Description

This PR adds the onboarding task list for the new homepage.
![Screenshot 2023-01-02 at 15 58 30](https://user-images.githubusercontent.com/1082140/210341232-5ff49e35-fdb4-493f-be85-d3df32b5a16a.png)


## Code review notes

The PR is a bit bigger because it also adds the store for the homepage app. I recommend reviewing commit by commit.

## QA notes

You need to activate the homepage feature on `/wp-admin/admin.php?page=mailpoet-experimental`
The logic for the steps is described in the ticket.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-4826]

## After-merge notes

_N/A_


[MAILPOET-4826]: https://mailpoet.atlassian.net/browse/MAILPOET-4826?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

costasovo committed:
Add basic task list component for homepage

[MAILPOET-4826]

costasovo committed:
Add dummy hide list button

[MAILPOET-4826]

costasovo committed:
Add an empty wordpress/data store to homepage

We plan to add more dynamic sections to the homepage so I want to set up and use the
store from the beginning.
[MAILPOET-4826]

costasovo committed:
Hide the task list and store the state in the store

[MAILPOET-4826]

costasovo committed:
Store homepage task list was dismissed to DB

[MAILPOET-4826]

costasovo committed:
Render simple task list items with state info

[MAILPOET-4826]

costasovo committed:
Add styles and links for homepage task list

[MAILPOET-4826]

costasovo committed:
Highlight current task in task list on homepage

[MAILPOET-4826]

costasovo committed:
Display WooCommerce import task based on customers count

We use the same logic in WooCommerce setup page
[MAILPOET-4826]

costasovo committed:
Add styles for task list heading

[MAILPOET-4826]

costasovo committed:
Add message when task list is complete

[MAILPOET-4826]

costasovo committed:
Add complete state task titles and additional task contents

[MAILPOET-4826]

costasovo committed:
Add basic acceptance test for the homepage task list

[MAILPOET-4826]

costasovo committed:
Add error boundaries to the homepage app

[MAILPOET-4826]

lysyjan reviewed (dismissed):
Good job!

Aschepikov commented:
Hi @costasovo  
<img width="485" alt="image" src="https://user-images.githubusercontent.com/113185437/211317152-3c0c1b56-9934-40d8-84cf-54fac8691072.png">
here in the ticket, we have that the link shouldn't work after finishing of the task, but it is still working, can we change it?

Aschepikov commented:
And here are some conflicts, can you resolve them pls

costasovo committed:
Use nicer name for property that indicates complete task

[MAILPOET-4826]

costasovo committed:
Make completed tasks in task list not interact-able

[MAILPOET-4826]

costasovo commented:
> Hi @costasovo <img alt="image" width="485" src="https://user-images.githubusercontent.com/113185437/211317152-3c0c1b56-9934-40d8-84cf-54fac8691072.png"> here in the ticket, we have that the link shouldn't work after finishing of the task, but it is still working, can we change it?

Thank you for finding this @Aschepikov and sorry for the inconvenience. I must have missed that one.

@lysyjan I added two new commits addressing the issue and I rebased the branch to get rid of the conflicts.

Aschepikov reviewed (approved):
GJ!
